internal/stacksapi/stack_notification.go

```diff
@@ -4,33 +4,40 @@ import (
 	"context"
 	"fmt"
 	"net/http"
+	"time"
 )
 
-type CreateStackNotificationRequest struct {
-	StackKey string `json:"-"`      // The key to call the stack. Required.
-	JobUUID  string `json:"-"`      // The uuid of a job
-	Detail   string `json:"detail"` // Short notification message (max length 255)
+type StackNotification struct {
+	JobUUID   string     `json:"job_uuid"`
+	Detail    string     `json:"detail"`
+	Timestamp *time.Time `json:"timestamp,omitempty"`
```

Comment:
If you use `omitzero` instead of `omitempty`, then `encoding/json` will call the `IsZero` method of `time.Time` to figure out whether or not it is set.

Reply:
Good catch!